raiden/blockchain/filters.py

```diff
@@ -109,59 +110,55 @@ class StatelessFilter(LogFilter):
     Pass latest block_number to get_(new|all)_entries to avoid querying it
     """
 
-    def __init__(self, web3: Web3, filter_params: Dict[str, Any]) -> None:
+    def __init__(
+        self,
+        web3: Web3,
+        from_block: BlockNumber,
+        contract_address: ChecksumAddress,
+        topics: Optional[List[Optional[str]]],
+    ) -> None:
         super().__init__(web3, filter_id=None)
-        self.filter_params: Dict[str, BlockSpecification] = filter_params
-        self._last_block: BlockNumber = BlockNumber(-1)
-        self._lock = Semaphore()
 
-    def _do_get_new_entries(
-        self, from_block: BlockSpecification, to_block: BlockSpecification
-    ) -> List[BlockchainEvent]:
-        filter_params = self.filter_params.copy()
-        filter_params["fromBlock"] = from_block
-        filter_params["toBlock"] = to_block
+        # fix off-by-one
+        last_block = from_block - 1
 
-        log.debug("Querying StatelessFilter", from_block=from_block, to_block=to_block)
-        result = self.web3.eth.getLogs(filter_params)
-        self._last_block = block_specification_to_number(block=to_block, web3=self.web3)
-        return result
+        self.contract_address = contract_address
+        self.topics = topics
+        self.last_block = last_block
+        self._lock = Semaphore()
 
     def from_block_number(self) -> BlockNumber:
-        filter_from_number = block_specification_to_number(
-            block=self.filter_params.get("fromBlock", GENESIS_BLOCK_NUMBER), web3=self.web3
-        )
-        return BlockNumber(max(filter_from_number, self._last_block + 1))
```

Comment:
This madness was complete simplified, `fromBlock` is not optional anymore, so, the `get` is unnecessary, the `get_all_entries` interface was removed (it didn't add any value, the same result can be achieve by instantiating an instancing and giving the class the block number of `latest`), because of there is only one value instead of both `fromBlock` and `_last_block`, so the max is not necessary anymore.